File: code2040-stage3.php
<?php

function noPref($prefix, $array){
  $newList = [];
  foreach ($array as $element){
    if (strpos($element, $prefix) !== 0){
      $newList[] = $element;
    }
  }
  return $newList;
}

$request = new stdClass();
$request->token = "1yxife3KZE";
 
$json_data = json_encode($request);
 
$post = file_get_contents('http://challenge.code2040.org/api/prefix',null,stream_context_create(array(
    'http' => array(
        'protocol_version' => 1.1,
        'user_agent'       => 'Diana_Lamaute',
        'method'           => 'POST',
        'header'           => "Content-type: application/json\r\n".
                              "Connection: close\r\n" .
                              "Content-length: " . strlen($json_data) . "\r\n",
        'content'          => $json_data,
    ),
)));
 
if ($post){
	$prefix = json_decode($post)->result->prefix;
  $array = json_decode($post)->result->array;

  $res = noPref($prefix, $array);

	$newRequest = new stdClass();
	$newRequest->token = "1yxife3KZE";
	$newRequest->array = $res;
 
	$json_data = json_encode($newRequest);
 
	$post = file_get_contents('http://challenge.code2040.org/api/validateprefix',null,stream_context_create(array(
    'http' => array(
        'protocol_version' => 1.1,
        'user_agent'       => 'Diana_Lamaute',
        'method'           => 'POST',
        'header'           => "Content-type: application/json\r\n".
                              "Connection: close\r\n" .
                              "Content-length: " . strlen($json_data) . "\r\n",
        'content'          => $json_data,
    ),
)));
} 
else echo "POST failed";

File: code2040-stage3.py
#exclude prefix

import requests
import json

def noPref(prefix, fullList):
	newList = []
	pref = len(prefix)
	for item in fullList:
		if prefix not in item[:pref]:
			newList.append(item)
	return newList

url = "http://challenge.code2040.org/api/prefix"
data = {"token":"1yxife3KZE"}
headers = {'Content-type': 'application/json', 'Accept': 'text/plain'}
r = requests.post(url, data=json.dumps(data), headers=headers)

prefix = r.json()['result']['prefix']
arr = r.json()['result']['array']

result = noPref(prefix, arr)

url = "http://challenge.code2040.org/api/validateprefix"
data = {"token":"1yxife3KZE", "array":result}
headers = {'Content-type': 'application/json', 'Accept': 'text/plain'}
r = requests.post(url, data=json.dumps(data), headers=headers)